File: src/tests/test_insertionSort.py
from typing import List

class InsertionSort:
    def sort(self, arr: int, size: int) -> None:
        for i in range(1, size):
            key = None
            key = arr[i]
            j = None
            j = i - 1
            while j >= 0 and arr[j] > key:
                arr[j + 1] = arr[j]
                (j:=j-1)
            arr[j + 1] = key
    def __init__(self):
            pass
def main():
    arr = [0] * 100
    size = None
    size = 0
    print("Input the array length: ", sep='', end='')
    size = int(input())
    print("Input integers: ", sep='', end='')
    for i in range(0, size):
        arr[i] = int(input())
    print("Original array: ", sep='', end='')
    for i in range(0, size):
        print(arr[i], " ", sep='', end='')
    print("\n", sep='', end='')
    sorter = InsertionSort()
    sorter.sort(arr, size)
    print("Sorted array: ", sep='', end='')
    for i in range(0, size):
        print(arr[i], " ", sep='', end='')
    print("\n", sep='', end='')
    return 0

if __name__ == '__main__':
    main()

File: src/tests/test_insertionSort.cpp
#include <iostream>

class InsertionSort {
public:
    void sort(int* arr, int size) {
        for (int i = 1; i < size; ++i) {
            int key = arr[i];
            int j = i - 1;

            while (j >= 0 && arr[j] > key) {
                arr[j + 1] = arr[j];
                --j;
            }
            arr[j + 1] = key;
        }
    }
};

int main() {
    int arr[100];
    int size = 0;

    // 输入数组长度
    std::cout << "Input the array length: ";
    std::cin >> size;

    // 输入整数
    std::cout << "Input integers: ";
    for (int i = 0; i < size; i++) {
        std::cin >> arr[i];
    }

    std::cout << "Original array: ";
    for (int i = 0; i < size; ++i) {
        std::cout << arr[i] << " ";
    }
    std::cout << std::endl;

    InsertionSort sorter;
    sorter.sort(arr, size);

    std::cout << "Sorted array: ";
    for (int i = 0; i < size; ++i) {
        std::cout << arr[i] << " ";
    }
    std::cout << std::endl;

    return 0;
}
